cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java

```diff
@@ -492,45 +417,22 @@ void registerAdapters() {
     }
   }
 
-
-  private AdapterTypeInfo createAdapterTypeInfo(File file, Manifest manifest) {
+  // TODO: call configure on the application itself to get name and program type instead of requiring
+  //       them in the manifest
+  private ApplicationTemplateInfo createAppTemplateInfo(File file, Manifest manifest) {
     if (manifest != null) {
       Attributes mainAttributes = manifest.getMainAttributes();
 
-      String adapterType = mainAttributes.getValue(AdapterManifestAttributes.ADAPTER_TYPE);
-      String adapterProgramType = mainAttributes.getValue(AdapterManifestAttributes.ADAPTER_PROGRAM_TYPE);
-      String defaultAdapterProperties = mainAttributes.getValue(AdapterManifestAttributes.ADAPTER_PROPERTIES);
-
-      String sourceType = mainAttributes.getValue(AdapterManifestAttributes.SOURCE_TYPE);
-      String sinkType = mainAttributes.getValue(AdapterManifestAttributes.SINK_TYPE);
-      String defaultSourceProperties = mainAttributes.getValue(AdapterManifestAttributes.SOURCE_PROPERTIES);
-      String defaultSinkProperties = mainAttributes.getValue(AdapterManifestAttributes.SINK_PROPERTIES);
-
-      if (adapterType != null && sourceType != null && sinkType != null && adapterProgramType != null) {
-        return new AdapterTypeInfo(file, adapterType, Source.Type.valueOf(sourceType.toUpperCase()),
-                                   Sink.Type.valueOf(sinkType.toUpperCase()),
-                                   propertiesFromString(defaultSourceProperties),
-                                   propertiesFromString(defaultSinkProperties),
-                                   propertiesFromString(defaultAdapterProperties),
-                                   ProgramType.valueOf(adapterProgramType.toUpperCase()));
-      }
+      String name = mainAttributes.getValue(ApplicationTemplateManifestAttributes.NAME);
```

Comment:
Do we still need Manifest file with these info? If we have the restriction of only template-supported program per template?